### PR TITLE
Validate Token Expiration

### PR DESCRIPTION
**Pull request by Ranibm**
https://github.ibm.com/security-services/appid-project-management/issues/3892

### CODE REVIEW

**Comment by Ranibm** [comment]
should I update the Readme with documentation on the new flag?

**Comment by aal80** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- test/webapp-strategy-test.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ibm-cloud-security/appid-serversdk-nodejs/pullRequest?prid=4851188&bid=16096951)

**Comment by aal80** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- test/webapp-strategy-test.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ibm-cloud-security/appid-serversdk-nodejs/pullRequest?prid=4851188&bid=16096951)

**Comment by aal80** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- test/webapp-strategy-test.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ibm-cloud-security/appid-serversdk-nodejs/pullRequest?prid=4851188&bid=16096951)

**Comment by aal80** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- test/webapp-strategy-test.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ibm-cloud-security/appid-serversdk-nodejs/pullRequest?prid=4851188&bid=16096951)

File: lib/strategies/webapp-strategy.js [hunk 1]
```diff
@@ -262,8 +262,18 @@ function handleAuthorization(req, options = {}, strategy) {
...
 }
 
-function isAuthenticated(req) {
-	return req.session[WebAppStrategy.AUTH_CONTEXT];
+function isAuthenticated(req, options = {}) {
+	if (!req.session[WebAppStrategy.AUTH_CONTEXT]) {
+		return false;
+	}
+	
+	const allowExpiredTokensOnSession = options.hasOwnProperty("allowExpiredTokensOnSession") ?
```
Comment (rotembr) [inline lib/strategies/webapp-strategy.js]: lets do the opposite. (simply change the true to false)
make it validate by default and add not document this flag.
when publish SDK we will increase the Minor version.

Comment (Ranibm) [reply lib/strategies/webapp-strategy.js]: good idea

**Comment by aal80** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- test/webapp-strategy-test.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ibm-cloud-security/appid-serversdk-nodejs/pullRequest?prid=4851188&bid=16096951)

**Pull request merged**